front-end/src/app/shared/components/transaction-type-base/triple-transaction-type-base.component.ts

```diff
@@ -119,47 +120,21 @@ export abstract class TripleTransactionTypeBaseComponent
     payload.children[0].report_id = payload.report_id;
     payload.children[1].report_id = payload.report_id;
 
-    if (payload.transaction_type_identifier) {
-      const responseFromApi = this.writeToApi(payload);
-      responseFromApi.subscribe((transaction) => {
-        navigationEvent.transaction = this.transactionType?.updateParentOnSave ? payload : transaction;
-        this.navigateTo(navigationEvent);
-      });
-    }
+    this.processPayload(payload, navigationEvent);
   }
 
-  override handleNavigate(navigationEvent: NavigationEvent): void {
-    this.formSubmitted = true;
-
-    if (navigationEvent.action === NavigationAction.SAVE) {
-      if (
-        this.form.invalid ||
-        this.childForm.invalid ||
-        this.childForm_2.invalid ||
-        !this.transaction ||
-        !this.childTransaction ||
-        !this.childTransaction_2
-      ) {
-        return;
-      }
-
-      let confirmation$ = this.confirmWithUser(this.transaction, this.form);
-      confirmation$ = concat(
-        confirmation$,
-        this.confirmWithUser(this.childTransaction, this.childForm, 'childDialog')
-      ).pipe(reduce((accumulator, confirmed) => accumulator && confirmed));
-      confirmation$ = concat(
-        confirmation$,
-        this.confirmWithUser(this.childTransaction_2, this.childForm_2, 'childDialog_2')
-      ).pipe(reduce((accumulator, confirmed) => accumulator && confirmed));
-
-      confirmation$.subscribe((confirmed: boolean) => {
-        // if every confirmation was accepted
-        if (confirmed) this.save(navigationEvent);
-      });
-    } else {
-      this.navigateTo(navigationEvent);
-    }
+  override isInvalid(): boolean {
+    return super.isInvalid() ||
+      this.childForm_2.invalid ||
+      !this.childTransaction_2;
+  }
+
+  override get confirmation$(): Observable<boolean> {
```

Comment:
WOW! This is a really impressive way of declaring the confirmations for each part once AND in their own component.